src/cli/Create.cpp

```diff
@@ -0,0 +1,173 @@
+ */
+int Create::execute(const QStringList& arguments)
+{
+    QTextStream out(stdout, QIODevice::WriteOnly);
+    QTextStream err(stdout, QIODevice::WriteOnly);
+
+    QCommandLineParser parser;
+
+    parser.setApplicationDescription(description);
+    parser.addPositionalArgument("database", QObject::tr("Path of the database."));
+    QCommandLineOption keyFile(
+        QStringList() << "k" << "key-file",
+        QObject::tr("Key file of the database. When file does not exist, new key file will be generated"),
+        QObject::tr("path"));
+    parser.addOption(keyFile);
+
+    parser.addHelpOption();
+    parser.process(arguments);
+
+    const QStringList args = parser.positionalArguments();
+    if (args.size() < 1) {
+        out << parser.helpText().replace("keepassxc-cli", "keepassxc-cli create");
+        return EXIT_FAILURE;
+    }
+
+    auto key = QSharedPointer<CompositeKey>(new CompositeKey());
+
+    QSharedPointer<FileKey> fileKey;
+    if (!loadFileKey(parser.value(keyFile), fileKey)) {
+        err << QObject::tr("Loading the key file failed") << endl;
+        return EXIT_FAILURE;
+    }
+
+    if (!fileKey.isNull()) {
+        key->addKey(fileKey);
+    }
+
+    auto password = getPasswordFromStdin();
+    if (!password.isNull()) {
+        key->addKey(password);
+    }
+
+    if (key->isEmpty()) {
+        err << QObject::tr("No key's set, aborting creating database") << endl;
+        return EXIT_FAILURE;
+    }
+
+    Database db;
+    db.setKey(key);
+
+    QString dberror = db.saveToFile(args.at(0));
+    if (dberror.length() > 0) {
+        err << QObject::tr("Failed to save the database: ") << dberror << endl;
+        return EXIT_FAILURE;
+    }
+
+    return EXIT_SUCCESS;
+}
+
+/**
+ * Read optional password from stdin.
+ *
+ * @return Pointer to the PasswordKey or null if passwordkey is skipped
+ *         by user
+ */
+QSharedPointer<PasswordKey> Create::getPasswordFromStdin()
+{
+    QSharedPointer<PasswordKey> passwordKey;
+    QTextStream out(stdout, QIODevice::WriteOnly);
+
+    out << QObject::tr("Insert password used to encrypt database (Press enter leave blank): ");
+    out.flush();
+    QString password = Utils::getPassword();
+
+    if (password.length() > 0) {
+        passwordKey = QSharedPointer<PasswordKey>(new PasswordKey(password));
+    }
+
+    return passwordKey;
+}
+
+/**
+ * Load a key file from disk. When the path specified does not exist a
+ * new file will be generated. No folders will be generated so the parent
+ * folder of the specified file nees to exist
+ *
+ * If the key file cannot be loaded or created the function will fail.
+ *
+ * @param path Path to the key file to be loaded
+ * @param fileKey Resulting fileKey
+ * @return true if the key file was loaded succesfully
+ */
+bool Create::loadFileKey(QString path, QSharedPointer<FileKey>& fileKey)
+{
+    QTextStream err(stdout, QIODevice::WriteOnly);
+
+    if (path.length() > 0) {
```

Comment:
See comment above about hidden magic. If you still want to keep this as an additional safeguard, then invert the logic and return if path.isEmpty() is true.

Reply:
Removed check on path